docs/docs/api-reference/core/RecoilRoot.md

```diff
@@ -5,6 +5,8 @@ sidebar_label: <RecoilRoot />
 
 Provides the context in which atoms have values. Must be an ancestor of any component that uses any Recoil hooks. Multiple roots may co-exist; atoms will have distinct values within each root. If they are nested, the innermost root will completely mask any outer roots.
 
+Note: This does **not** mean that `RecoilRoot`s are always completely isolated from each other. For example, a selector that does not depend on any atom may have its result cached across roots. This can cause problems if the selector reads external resources, especially in tests that only render a new `RecoilRoot` for every test, but do not [reset](/docs/guides/asynchronous-data-queries#use-a-request-id) the selector in any way.
```

Comment:
Adding some clarification may be helpful, but some corrections here:  It does not matter if the selectors have dependencies on other atoms/selectors or not for this.  It is also not a *problem* if the user is properly following the API contract when implementing their selectors to be properly "idempotent" (returning the same value for the same set of inputs/dependencies with no side-effects except for caching/logging).  Also, the term "reset" means to reset an atom/selector to the default value, which is distinct from the "refresh" pattern described in the guide.

Reply:
I don't view the first two points as corrections.

> It does not matter if the selectors have dependencies on other atoms/selectors or not for this.

The referenced sentence starts with "For example" - it does not claim that there are no other situations in which this occurs, such as when a selector depends on an atom but the atom happens to have the same value across roots.

> It is also not a _problem_

It is not stated that Recoil's caching is a problem, it is stated that Recoil's caching in conjunction with certain user code can *cause* problems. This does not make a statement about either the former or the latter being at fault.

I agree that the wording reset is unfortunate, given that Recoil already uses it to describe one of its concepts. Substituting it with "refresh" would make sense and leave the meaning of the sentence intact.
Let me know if you want me to make this or other minor changes, however if as it appears you disagree with the phrasing of the paragraph overall, it'd probably be better if you rephrase it yourself from a maintainer POV, since it's barely worth bikeshedding any more over this few words :sweat_smile: